framework/src/modules/dpos/transaction_assets/pom_transaction_asset.ts

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { BaseBlockHeader } from '@liskhq/lisk-chain';
+import { getAddressFromPublicKey } from '@liskhq/lisk-cryptography';
+import { codec } from '@liskhq/lisk-codec';
+import { ApplyAssetInput, BaseAsset, ValidateAssetInput } from '../../base_asset';
+import { ValidationError } from '../../../errors';
+import { MAX_PUNISHABLE_BLOCK_HEIGHT_DIFFERENCE, MAX_POM_HEIGHTS } from '../constants';
+import { DPOSAccountProps, PomTransactionAssetInput } from '../types';
+import { getPunishmentPeriod, validateSignature } from '../utils';
+
+const signingBlockHeaderSchema = {
+	$id: 'lisk/signing-block-header',
+	type: 'object',
+	properties: {
+		version: { dataType: 'uint32', fieldNumber: 1 },
+		timestamp: { dataType: 'uint32', fieldNumber: 2 },
+		height: { dataType: 'uint32', fieldNumber: 3 },
+		previousBlockID: { dataType: 'bytes', fieldNumber: 4 },
+		transactionRoot: { dataType: 'bytes', fieldNumber: 5 },
+		generatorPublicKey: { dataType: 'bytes', fieldNumber: 6 },
+		reward: { dataType: 'uint64', fieldNumber: 7 },
+		asset: {
```

Comment:
shouldn't asset be type bytes?
if so then we could use `signingBlockHeaderSchema` from `lisk-chain`

Reply:
It was already there in POM transaction, and I think its correct. The header1 and header2 which are passed to dpos asset are in object form not as encoded buffer.